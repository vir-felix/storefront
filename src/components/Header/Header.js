import React, { Component } from "react";
import PropTypes from "prop-types";
import AppBar from "material-ui/AppBar";
import Hidden from "material-ui/Hidden";
import MenuList from "material-ui/Menu/MenuList";
import Toolbar from "material-ui/Toolbar";
import Typography from "material-ui/Typography";
import IconButton from "material-ui/IconButton";
import Drawer from "material-ui/Drawer";
import CartIcon from "mdi-material-ui/Cart";
import MenuIcon from "mdi-material-ui/Menu";
import Popover from "material-ui/Popover";
import { inject, observer } from "mobx-react";
import { action, observable, computed } from "mobx";
import { withStyles } from "material-ui/styles";
import NavigationItem from "../NavigationItem";
import HorizontalNavigationItem from "../NavigationItem/HorizontalNavigationItem";

const styles = () => ({
  cart: {
    width: 320
  },
  menu: {
    flex: 1
  }
});

// TODO: Get tag data from GraphQL, this is just a sample
const tags = [
  { name: "men", title: "Men" },
  { name: "women", title: "Women" },
  {
    name: "kids",
    title: "Kids",
    relatedTags: [
      {
        name: "test1",
        title: "Test 1",
        relatedTags: [
          { name: "men", title: "Men" },
          { name: "women", title: "Women" },
          { name: "men", title: "Men" },
          { name: "women", title: "Women" }
        ]
      },
      {
        name: "test2",
        title: "Test 2",
        relatedTags: [
          { name: "men", title: "Men" },
          { name: "women", title: "Women" },
          { name: "men", title: "Men" },
          { name: "women", title: "Women" }
        ]
      },
      { name: "test3", title: "Test 3" }
    ]
  }
];

@withStyles(styles)
@inject("uiStore")
@observer
class Header extends Component {
  static propTypes = {
    classes: PropTypes.object,
    uiStore: PropTypes.object
  };

  @observable _popoverAnchor = null;

  @computed
  get popoverAnchor() {
    return this._popoverAnchor;
  }

  @action
  onClick = event => {
    this._popoverAnchor = event.target;
  };

  @action
  onClose = () => {
    this._popoverAnchor = null;
  };

  render() {
    const { classes, uiStore } = this.props;

    return (
      <AppBar position="static" elevation={0}>
        <Toolbar>
          <Hidden mdUp>
            <IconButton color="inherit" onClick={uiStore.toggleMenuDrawerOpen}>
              <MenuIcon />
            </IconButton>
          </Hidden>

          <Typography className={classes.title} color="inherit" variant="title">
            Reaction
          </Typography>

          <nav className={classes.menu}>
            <Hidden smDown>{tags.map((tag, index) => <HorizontalNavigationItem key={index} menuItem={tag} />)}</Hidden>
          </nav>

          <IconButton color="inherit" onClick={uiStore.toggleCartOpen}>
            <CartIcon />
          </IconButton>
        </Toolbar>
        <Drawer open={uiStore.menuDrawerOpen} onClose={uiStore.toggleMenuDrawerOpen}>
          <div className={classes.cart}>
            <MenuList>{tags.map((tag, index) => <NavigationItem key={index} menuItem={tag} />)}</MenuList>
          </div>
        </Drawer>
        <Drawer anchor="right" open={uiStore.cartOpen} onClose={uiStore.toggleCartOpen}>
          <div className={classes.cart}>Cart</div>
        </Drawer>
        <Popover anchorEl={this.popoverAnchor} onClose={this.onClose} open={Boolean(this.popoverAnchor)}>
          <Typography variant="title">Popover!</Typography>
        </Popover>
      </AppBar>
    );
  }
}

export default Header;
