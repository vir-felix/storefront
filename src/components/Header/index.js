import React, { Component } from "react";
import AppBar from "material-ui/AppBar";
import Toolbar from "material-ui/Toolbar";
import Typography from "material-ui/Typography";

class Header extends Component {
  render() {
    return (
      <AppBar position="static" color="primary">
        <Toolbar>
          <Typography variant="title" color="secondary">App Name</Typography>
        </Toolbar>
      </AppBar>
    );
  }
}

export default Header;
