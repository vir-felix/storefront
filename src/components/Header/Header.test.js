
import React from "react";
import renderer from "react-test-renderer";
import Header from "./Header";

test("basic snapshot", () => {
  const component = renderer.create(<Header />);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
