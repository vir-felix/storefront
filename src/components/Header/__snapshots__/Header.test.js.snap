// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic snapshot 1`] = `
<header
  className="MuiPaper-root-9 MuiPaper-elevation4-15 MuiAppBar-root-1 MuiAppBar-positionStatic-5 MuiAppBar-colorPrimary-7"
>
  <div
    className="MuiToolbar-root-36 MuiToolbar-gutters-37"
  >
    <h2
      className="MuiTypography-root-38 MuiTypography-title-44"
    >
      Reaction Commerce
    </h2>
  </div>
</header>
`;
